lib/hypernova/batch_renderer.rb

```diff
@@ -59,8 +63,9 @@ def render_blank
 
   attr_reader :jobs
 
-  def extract_html_from_result(name_of_component, result)
-    result["html"].nil? ? render_blank_html(jobs[name_of_component]) : result["html"]
+  def ensure_has_html(name_of_component, result)
+    result['html'] = result['html'].nil? ? render_blank_html(jobs[name_of_component]) : result['html']
```

Comment:
`result['html'] = render_blank_html(jobs[name_of_component]) if result['html'].nil?` to avoid the unconditional assignment